internal/pkg/values/timespan.go

```diff
@@ -143,14 +145,58 @@ func (t relativeTimeSpan) isTimeOfDayInRange(timeOfDay time.Time) bool {
 	return (t.timeFrom.Before(timeOfDay) || t.timeFrom.Equal(timeOfDay)) && t.timeTo.After(timeOfDay)
 }
 
-// isTimeInSpan check if the time is in the span
+// isTimeInSpan checks if the time is in the span
 func (t relativeTimeSpan) isTimeInSpan(targetTime time.Time) bool {
 	targetTime = targetTime.In(t.timezone)
 	timeOfDay := getTimeOfDay(targetTime)
 	weekday := targetTime.Weekday()
 	return t.isTimeOfDayInRange(timeOfDay) && t.isWeekdayInRange(weekday)
 }
 
+// inLocation returns an array of relative timespans matching the timespan converted to the given location
+func (t relativeTimeSpan) inLocation(timezone *time.Location) []relativeTimeSpan {
+	var result []relativeTimeSpan
+	sameDays := relativeTimeSpan{
+		timezone:    timezone,
+		weekdayFrom: t.weekdayFrom,
+		weekdayTo:   t.weekdayTo,
+		timeFrom:    t.timeFrom.In(timezone),
+		timeTo:      t.timeTo.In(timezone),
+	}
+	result = append(result, sameDays)
+	if isTimeFromSkippedToPreviousDay(sameDays.timeFrom) {
+		daysBefore := relativeTimeSpan{
+			timezone:    timezone,
+			timeFrom:    sameDays.timeFrom.Add(24 * time.Hour),
+			timeTo:      sameDays.timeTo.Add(24 * time.Hour),
+			weekdayFrom: sameDays.weekdayFrom - 1,
+			weekdayTo:   sameDays.weekdayTo - 1,
+		}
+		result = append(result, daysBefore)
+	}
+	if isTimeToSkippedToNextDay(sameDays.timeTo) {
+		daysAfter := relativeTimeSpan{
+			timezone:    timezone,
+			timeFrom:    sameDays.timeFrom.Add(-24 * time.Hour),
+			timeTo:      sameDays.timeTo.Add(-24 * time.Hour),
+			weekdayFrom: sameDays.weekdayFrom + 1,
+			weekdayTo:   sameDays.weekdayTo + 1,
+		}
+		result = append(result, daysAfter)
+	}
+	return result
+}
+
+// Check if timeFrom skipped to the previous day
+func isTimeFromSkippedToPreviousDay(timeFrom time.Time) bool {
```

Comment:
i would change this to `hasTimeFromSkippedToPreviousDay` and maybe put it in util.go.
Also the comment should be in this format: `// functionName description`

Same with `isTimeToSkippedToNextDay`.